dashboard/src/pages/TreeDetails/TreeDetailsFilter.tsx

```diff
@@ -0,0 +1,260 @@
+} from '@/components/Filter/CheckboxSection';
+import {
+  TFilter,
+  TFilterKeys,
+  TreeDetails as TreeDetailsType,
+} from '@/types/tree/TreeDetails';
+import { useTreeDetails } from '@/api/TreeDetails';
+import { Skeleton } from '@/components/ui/skeleton';
+
+import { cleanFalseFilters } from './treeDetailsUtils';
+
+const zFilterValue = z.record(z.boolean());
+type TFilterValues = z.infer<typeof zFilterValue>;
+
+interface ITreeDetailsFilter {
+  filter: TFilter;
+  treeUrl: string;
+}
+
+const filterFieldMap = {
+  git_repository_branch: 'branches',
+  config_name: 'configs',
+  architecture: 'archs',
+  valid: 'status',
+} as const satisfies Record<string, TFilterKeys>;
+
+export const mapFilterToReq = (
+  filter: TFilter,
+): { [key: string]: string[] } => {
+  const filterMapped: { [key: string]: string[] } = {};
+
+  Object.entries(filterFieldMap).forEach(([reqField, field]) => {
+    const values = filter[field];
+    if (!values) return;
+
+    Object.entries(values).forEach(([value, isSelected]) => {
+      if (isSelected) {
+        if (reqField == 'valid') {
+          value = value == 'Sucess' ? 'true' : 'false';
+        }
+        if (!filterMapped[reqField]) filterMapped[reqField] = [];
+        filterMapped[reqField].push(value);
+      }
+    });
+  });
+
+  return filterMapped;
+};
+
+export const createFilter = (data: TreeDetailsType | undefined): TFilter => {
+  const status = { Sucess: false, Failure: false };
+  const branches: TFilterValues = {};
+  const configs: TFilterValues = {};
+  const archs: TFilterValues = {};
+
+  if (data)
+    data.builds.forEach(b => {
+      if (b.git_repository_branch) branches[b.git_repository_branch] = false;
+      if (b.config_name) configs[b.config_name] = false;
+      if (b.architecture) archs[b.architecture] = false;
+    });
+
+  return { status, branches, configs, archs };
+};
+
+const changeFilterValue = (
+  filter: TFilter,
+  filterField: TFilterKeys,
+  value: string,
+): TFilter => {
+  const newFilter: TFilter = JSON.parse(JSON.stringify(filter ?? {}));
+  if (!newFilter[filterField]) {
+    newFilter[filterField] = {};
+  }
+  const filterSection = newFilter[filterField];
+  const filterValue = filterSection[value] ?? false;
+  filterSection[value] = !filterValue;
+
+  return newFilter;
+};
+
+type CheckboxSectionProps = {
+  diffFilter: TFilter;
+  setDiffFilter: Dispatch<SetStateAction<TFilter>>;
+};
+
+// TODO: Remove useState for this forms, use something like react hook forms or tanstack forms (when it gets released)
+const CheckboxSection = ({
+  diffFilter,
+  setDiffFilter,
+}: CheckboxSectionProps): JSX.Element => {
+  const intl = useIntl();
+
+  const { treeId } = useParams({ from: '/tree/$treeId/' });
+
+  const { data, isLoading } = useTreeDetails(treeId);
+
+  const filter: TFilter = useMemo(() => {
+    return createFilter(data);
+  }, [data]);
+
+  const parsedFilter: TFilter = useMemo(() => {
+    const result = structuredClone(filter);
```

Comment:
sometimes you use `structuralClone` and other times `JSON.parse(JSON.stringify(obj))`, is there any reason for this "inconsistency"?

Reply:
no reason, just autopilot